ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/UpdatedMetaDataTracker.java

```diff
@@ -0,0 +1,76 @@
+/**
```

Comment:
It's not possible in all the cases especially InsertHandler which is piggy-back on TableHandler which unnecessarily add extra tasks.
Also, need to push the code of addDependants to precursor into each handlers.
So, thinking, not to move it in.